crowbar_framework/lib/crowbar/upgrade_status.rb

```diff
@@ -97,7 +97,7 @@ def finished?
 
     def save
       progress_file_path.open("w") do |f|
-        f.write(JSON.pretty_generate(progress))
+        f.write(YAML.dump(progress))
```

Comment:
Test suite might be working until now, but library is not, because at some point it checks for strings instead of symbols or vice versa. It is fixed by either deep_symbolize_keys call or by moving to YAML, but it is not specific to thread safe topic